src/payment/strategies/stripe-upe/format-locale.ts

```diff
@@ -0,0 +1,12 @@
+import { supportedLocales } from './stripe-upe-supported-locales';
+
+export default function formatLocale(localeLanguage: string): string {
+    const [language, country] = localeLanguage.replace(/_/g, '-').toLowerCase().split('-');
+    const countryLocales = supportedLocales[language];
+    if (!countryLocales) {
+        return 'auto';
+    }
+    const formattedLocale = `${language}-${country?.toUpperCase()}`;
```

Comment:
In this case if language is `en` and country is undefined, then the formatted locale will be `en-undefined`. Maybe we address this in a better way?

Reply:
In that case it would be `en-` because country has the optional chaining operator, and you cant call toUpperCase on undefined, and even then it doesn't matter because in the sentence below we check if this formattedLocale is in the supportedLocales array.
`en-` won't be in the array so it will return `countryLocales[0]` which in this case is `en`

Reply:
Sorry, forgot the backtick will transform the undefined into a string, but the second part of my comment still holds